src/model/consent/consent.ts

```diff
@@ -42,45 +42,62 @@
  * of 'createdAt' field may need to be tested in the future.
  */
 
-import { NotFoundError, RevokedConsentModificationError } from '../errors'
-import Knex from 'knex'
+import { NotFoundError } from '../errors'
+import Knex from 'knex';
 
 /*
  * Interface for Consent resource type
  */
 export interface Consent {
   id: string;
-  initiatorId?: string;
-  participantId?: string;
+  // participant DFSP that requested a Consent resource be made
+  participantId: string;
+  // status of the consent
   status: string;
-  credentialId?: string;
-  credentialType?: string;
-  credentialStatus?: string;
-  credentialPayload?: string;
-  credentialChallenge?: string;
+  // NOTE: not sure what purpose credentialId serves.
+  credentialId: string;
+  // credential type - currently trying to support FIDO/Generic credentials
+  credentialType: 'FIDO' | 'GENERIC' | undefined;
+  // NOTE: unsure this field is needed since `auth-service` verifies
+  //       credential on receiving them
+  credentialStatus: string;
+  // assuming this is the public key of the pair
+  credentialPayload: string;
+  // NOTE: not sure what `credentialChallenge` is used for.
+  //       best guess is that in `retrieveValidConsent` the original challenge
+  //       used to register the consent is needed to retrieve the Consent.
+  //       assuming this is an extra layer of protection since only the
+  //       DFSP has the variables used to derive the original challenge?
+  // TODO: remove this if it serves a redundant purpose
+  credentialChallenge: string;
```

Comment:
This is the original challenge sent by the DFSP to the PISP that is derived from the `scopes` + `consentId`: https://github.com/mojaloop/pisp-project/tree/master/docs/linking#161-deriving-the-challenge 

I think we should keep this as proof that the credential is indeed valid based on the `credentialPayload`

Reply:
sounds good